common/djangoapps/embargo/api.py

```diff
@@ -0,0 +1,242 @@
+"""
+The Python API layer of the country access settings. Essentially the middle tier of the project, responsible for all
+business logic that is not directly tied to the data itself.
+
+This API is exposed via the middleware(emabargo/middileware.py) layer but may be used directly in-process.
+
+"""
+
+from functools import partial
+import logging
+import pygeoip
+
+from django.core.cache import cache
+from django.conf import settings
+from django.http import HttpResponseRedirect, HttpResponseForbidden
+
+from student.models import unique_id_for_user
+from embargo.models import CountryAccessRule, IPFilter, RestrictedCourse, WHITE_LIST, BLACK_LIST
+
+log = logging.getLogger(__name__)
+
+# Reasons a user might be blocked.
+# These are used to generate info messages in the logs.
+REASONS = {
+    "ip_blacklist": u"Restricting IP address {ip_addr} {from_course} because IP is blacklisted.",
+    "ip_country": u"Restricting IP address {ip_addr} {from_course} because IP is from country {ip_country}.",
+    "profile_country": (
+        u"Restricting user {user_id} {from_course} because "
+        u"the user set the profile country to {profile_country}."
+    )
+}
+
+
+def _from_course_msg(course_id, course_is_embargoed):
+    """
+    Format a message indicating whether the user was blocked from a specific course.
+    This can be used in info messages, but should not be used in user-facing messages.
+
+    Args:
+        course_id (unicode): The ID of the course being accessed.
+        course_is_embarged (boolean): Whether the course being accessed is embargoed.
+
+    Returns:
+        unicode
+
+    """
+    return (
+        u"from course {course_id}".format(course_id=course_id)
+        if course_is_embargoed
+        else u""
+    )
+
+
+def _log_embargo_reason(check_func, course_id, course_is_embargoed):
```

Comment:
As you're cleaning up the `check_access` function, some of these helpers will no longer be necessary and can be deleted.